server/aesdsocket.c

```diff
@@ -233,55 +287,93 @@ void *thread_function(void *thread_param) {
                 //Then close the fd
                 close(temp_fd);
             } else {
-                //Open it in Read and Writing mode
-                int temp_fd = open(temp_file, O_CREAT|O_RDWR,0666);
-                //Check temp file return code
-                if (temp_fd == -1) {
-                    perror("Error opening temporary file for copying");
-                    exit(EXIT_FAILURE);
+
+                struct aesd_seekto seek_params;
+                //We check the current temp file to see if the string matches our command
+                bool ioctl_called = check_for_ioctl_cmd(temp_file, &(seek_params.write_cmd),
```

Comment:
Could just pass &seek_params here since the function needs to fill in the entire structure anyway.